x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterLookupJoinIT.java

```diff
@@ -557,6 +561,42 @@ public void testLookupJoinFieldTypes() throws IOException {
         }
     }
 
+    public void testLookupJoinRetryAnalysis() throws IOException {
+        setupClusters(3);
+        setSkipUnavailable(REMOTE_CLUSTER_1, false);
+        setSkipUnavailable(REMOTE_CLUSTER_2, false);
+
+        var defaultSettings = Settings.builder();
+        createIndexWithDocument(LOCAL_CLUSTER, "data", defaultSettings, Map.of("key", 1, "f1", 1));
+        createIndexWithDocument(REMOTE_CLUSTER_1, "data", defaultSettings, Map.of("key", 2, "f2", 2));
+        createIndexWithDocument(REMOTE_CLUSTER_2, "data", defaultSettings, Map.of("key", 3, "f3", 3));
+
+        var lookupSettings = Settings.builder().put(IndexSettings.MODE.getKey(), IndexMode.LOOKUP);
+        createIndexWithDocument(LOCAL_CLUSTER, "lookup", lookupSettings, Map.of("key", 1, "location", "local"));
+        createIndexWithDocument(REMOTE_CLUSTER_1, "lookup", lookupSettings, Map.of("key", 2, "location", "remote-1"));
+        // lookup is intentionally absent on REMOTE_CLUSTER_2
+
+        // The following query uses filter f2=2 that narrows down execution only to REMOTE_CLUSTER_1 index however,
+        // later it uses `WHERE f1 == 1` esql condition that to an attribute present only on the local cluster index.
+        // This causes analysis to fail and retry the entire query without a filter.
+        // The second analysis executes against all cluster indices and should discover that lookup is absent on REMOTE_CLUSTER_2.
+        expectThrows(
+            VerificationException.class,
+            containsString("lookup index [lookup] is not available in remote cluster [remote-b]"),
+            () -> runQuery(
+                syncEsqlQueryRequest().query("FROM data,*:data | LOOKUP JOIN lookup ON key | WHERE f1 == 1")
+                    .filter(new TermQueryBuilder("f2", 2))
+            )
+        );
```

Comment:
This test would fail if we only retry previously resolved subset of clusters opposed to ones resolved to all clusters without the filter.